A coffee shop claims that their coffee is served at a mean temperature of 170 degrees Fahrenheit. You suspect that the coffee is not as hot as they claim. You collect a sample of 20 cups of coffee and measure their temperatures. The sample mean temperature is 167 degrees Fahrenheit with a standard deviation of 3 degrees. Perform a hypothesis test at a 5% significance level to determine whether the coffee is served at a temperature different from 170 degrees Fahrenheit.


In [ ]:
import scipy.stats as stats

H0 = "Coffee served at 170 degree"
H1 = "Coffee not served at 170 degree"

pop_mean  = 170
sample_mean = 167
sample_std = 3
sample_size = 20
alpha = 0.05

# Calculate the standard error of the mean
standard_error = pop_mean / (sample_size ** 0.5)

# Calculate the t-statistic
t_statistic = (sample_mean - pop_mean) / standard_error

p_value = stats.t.cdf(t_statistic, df=sample_size-1)

if p_value < alpha:
    print("Reject the null hypothesis (HO)")
else:
    print("Fail to reject the null hypothesis (H1)")


print("p_value is : ",p_value)

Fail to reject the null hypothesis (H1)
p_value is :  0.4689607422661397


In [ ]:
marks = [587, 602, 627, 610, 619, 622, 605, 608, 596, 592]
pop_mean=600

H0 = "Girls score more than 600"
H1 ="Girls do not score more than 600"

t_val,p_val = stats.ttest_1samp(marks,pop_mean)
alpha = 0.05

print(p_val)
if p_val<alpha:
  print('Reject null Hypothesis : Girls score more than 600')
else:
  print('Fail to Reject null Hypothesis : Girls do not score more than 600')

0.1361270069204197
Fail to Reject null Hypothesis : Girls do not score more than 600


A car manufacturer claims that their new model has an average fuel efficiency of 30 miles per gallon (mpg). You collect a random sample of 25 cars and measure their fuel efficiency. The sample mean is 28.5 mpg, and the sample standard deviation is 3 mpg. Test the claim at a 1% significance level.

One sample t-test
Data:
Systolic blood pressures of 14 patients are given below:

183, 152, 178, 157, 194, 163, 144, 114, 178, 152, 118, 158, 172, 138

Test, whether the population mean, is less than 165

paired t-test
Data:
Eleven schoolboys were given a test in Statistics. They were given a Month’s tuition and a second test were held at the end of it. Do the marks give evidence that the students have benefited from the exam coaching?

Marks in 1st test: 23 20 19 21 18 20 18 17 23 16 19

Marks in 2nd test: 24 19 22 18 20 22 20 20 23 20 18

Hypothesis
H0: The students have not benefited from the tuition class. i.e., d = 0

H1: The students have benefited from the tuition class. i.e., d < 0

Where, d = x-y; d is the difference between marks in the first test (say x) and marks in the second test (say y).

Test statistic

Where, n is the number of samples ‘s’ is sample standard deviation

In [ ]:
alpha = 0.05
first_test =[23, 20, 19, 21, 18, 20, 18, 17, 23, 16, 19]
second_test=[24, 19, 22, 18, 20, 22, 20, 20, 23, 20, 18]

t_value,p_value=stats.ttest_rel(first_test,second_test)

one_tailed_p_value=float("{:.6f}".format(p_value/2))

print('Test statistic is %f'%float("{:.6f}".format(t_value)))

print('p-value for one_tailed_test is %f'%one_tailed_p_value)

alpha = 0.05

if one_tailed_p_value<=alpha:

    print('Conclusion','n','Since p-value(=%f)'%one_tailed_p_value,'<','alpha(=%.2f)'%alpha,'''We reject the null hypothesis H0.

So we conclude that the students have benefited by the tuition class. i.e., d = 0 at %.2f level of significance.'''%alpha)

else:

    print('Conclusion','n','Since p-value(=%f)'%one_tailed_p_value,'>','alpha(=%.2f)'%alpha,'''We do not reject the null hypothesis H0.

So we conclude that the students have not benefited by the tuition class. i.e., d = 0 at %.2f level of significance.'''%alpha)

Test statistic is -1.707331
p-value for one_tailed_test is 0.059282
Conclusion n Since p-value(=0.059282) > alpha(=0.05) We do not reject the null hypothesis H0. 

So we conclude that the students have not benefited by the tuition class. i.e., d = 0 at 0.05 level of significance.


#CHI-SQUARE TEST


Question: You have been observing the frequency of visits by different bird species at a bird feeder during a 24-hour period. You hypothesize that the visits are evenly distributed among four bird species: Sparrows, Robins, Blue Jays, and Cardinals. You want to test whether the observed frequencies match the expected frequencies based on your hypothesis.

You have collected the following data:

Observed Frequencies:
Frequency of visits by bird species at a bird feeder during a 24-hour period
Observed Frequencies:
Sparrows: 30 visits
Robins: 20 visits
Blue Jays: 25 visits
Cardinals: 25 visits

Your expected distribution (null hypothesis) is that each bird species visits the feeder equally (25% for each). You want to determine whether there is a significant difference between the observed and expected frequencies of bird visit


In [ ]:
import numpy as np
from scipy.stats import chisquare

# Observed frequencies
observed = np.array([30, 20, 25, 25])

# Expected frequencies based on the null hypothesis (even distribution)
expected = np.array([25, 25, 25, 25])

# Perform the goodness-of-fit chi-square test
chi2, p = chisquare(observed, f_exp=expected)

# Print the results
print("Chi-Square Statistic:", chi2)
print("P-value:", p)

Chi-Square Statistic: 2.0
P-value: 0.5724067044708798


In this case, the null hypothesis is that the visits by different bird species are evenly distributed, and the alternative hypothesis is that they are not evenly distributed. The chi-square test will help you determine whether the observed frequencies significantly differ from the expected frequencies, indicating whether the birds have a preference for certain times of the day.

 If the p-value for the goodness-of-fit chi-square test is 0.5724 (rounded to four decimal places), the result would typically be interpreted as follows:

Since the p-value is relatively high (greater than the common significance level of 0.05 or 5%), you would fail to reject the null hypothesis. In this context, it means that there is not enough evidence to conclude that the observed bird visit frequencies significantly differ from the expected frequencies based on an even distribution among the bird species.

In other words, the data does not provide sufficient support to suggest that the birds have a strong preference for specific times of the day at the bird feeder. The observed and expected frequencies are consistent with the hypothesis of an even distribution.

Question: Is there a significant association between a person's political affiliation and their preferred mode of transportation to work?

In [ ]:
import numpy as np
import pandas as pd
from scipy.stats import chi2_contingency

# Create a contingency table
data = pd.DataFrame({
    'PoliticalAffiliation': ['Democrat', 'Republican', 'Independent', 'Democrat', 'Independent', 'Republican'],
    'TransportationMode': ['Car', 'Public Transit', 'Bike', 'Car', 'Car', 'Public Transit']
})

contingency_table = pd.crosstab(data['PoliticalAffiliation'], data['TransportationMode'])

# Perform the chi-square test
chi2, p, dof, expected = chi2_contingency(contingency_table)

# Print the results
print("Chi-Square Statistic:", chi2)
print("P-value:", p)
print("Degrees of Freedom:", dof)
print("Expected Frequencies Table:")
print(expected)


Chi-Square Statistic: 8.000000000000002
P-value: 0.09157819444367084
Degrees of Freedom: 4
Expected Frequencies Table:
[[0.33333333 1.         0.66666667]
 [0.33333333 1.         0.66666667]
 [0.33333333 1.         0.66666667]]


In this example, the null hypothesis is that there is no association between political affiliation and transportation mode, and the alternative hypothesis is that they are associated. The p-value will help you determine if there is a significant association.

You can use the chi2_contingency function from the SciPy library to perform the chi-square test and obtain the test statistic, p-value, degrees of freedom, and expected frequencies. You can then interpret the results to determine whether there is a significant association between the variables.

Test of independance

You are conducting a survey to examine the relationship between a person's educational level and their choice of favorite leisure activity. You collect data from 200 respondents and observe the following distribution:

Educational Level:

High School: 50 , Bachelor's Degree: 60 , Master's Degree: 40 , PhD: 50 ,

Leisure Activity:

Reading: 70 , Sports: 45, Music: 60, Travel: 25

Perform a chi-square test of independence to determine whether there is a significant association between educational level and leisure activity preference. Use Python to conduct the test and interpret the results.

In [ ]:
import numpy as np
from scipy.stats import chi2_contingency

# Create a contingency table
data = np.array([[70, 45, 60, 25], [50, 60, 40, 50]])

# Perform the chi-square test of independence
chi2, p, dof, expected = chi2_contingency(data)

# Print the results
print("Chi-Square Statistic:", chi2)
print("P-value:", p)
print("Degrees of Freedom:", dof)
print("Expected Frequencies Table:")
print(expected)


Chi-Square Statistic: 17.80952380952381
P-value: 0.0004814800607489958
Degrees of Freedom: 3
Expected Frequencies Table:
[[60.  52.5 50.  37.5]
 [60.  52.5 50.  37.5]]


Question 2: Goodness of Fit

You are a wildlife biologist studying the distribution of bird species at a specific location in a forest. Based on your prior research, you hypothesize that four bird species (A, B, C, D) should be equally distributed in the area. You collect data from a sample of 200 bird sightings and observe the following distribution:

Species A: 55 sightings
Species B: 45 sightings
Species C: 50 sightings
Species D: 50 sightings
Perform a chi-square goodness-of-fit test using Python to determine whether the observed distribution matches the expected distribution based on your hypothesis. Use a significance level of 0.05.

In [ ]:
import numpy as np
from scipy.stats import chisquare

# Observed frequencies
observed = np.array([55, 45, 50, 50])

# Expected frequencies based on the null hypothesis (even distribution)
expected = np.array([50, 50, 50, 50])

# Perform the goodness-of-fit chi-square test
chi2, p = chisquare(observed, f_exp=expected)

# Significance level (alpha)
alpha = 0.05

# Print the results and make a decision
print("Chi-Square Statistic:", chi2)
print("P-value:", p)

if p <= alpha:
    print("Reject the null hypothesis. The observed distribution is significantly different from the expected distribution.")
else:
    print("Fail to reject the null hypothesis. The observed distribution matches the expected distribution.")


Chi-Square Statistic: 1.0
P-value: 0.8012519569012009
Fail to reject the null hypothesis. The observed distribution matches the expected distribution.


Question 2: Goodness of Fit

Scenario: You are investigating the distribution of smartphone brands used by a sample of 200 individuals. You hypothesize that the distribution of smartphone brands in the population should follow a certain distribution: Apple (30%), Samsung (40%), Google (20%), Other (10%). Perform a chi-square goodness-of-fit test to determine whether the observed distribution matches your hypothesis.

In [ ]:
import numpy as np
from scipy.stats import chisquare

# Observed frequencies
observed = np.array([60, 80, 35, 25])  # Counts for Apple, Samsung, Google, Other

# Expected frequencies based on the null hypothesis (hypothesized distribution)
expected = np.array([0.3 * 200, 0.4 * 200, 0.2 * 200, 0.1 * 200])

# Perform the chi-square goodness-of-fit test
chi2, p = chisquare(observed, f_exp=expected)

# Print the results
print("Chi-Square Statistic:", chi2)
print("P-value:", p)


Chi-Square Statistic: 1.875
P-value: 0.5987516330675617
